### PR TITLE
Add api function to logout, clean up the user data after logout

### PR DESCRIPTION
**Pull request by maryna-velychko**

### CODE REVIEW

File: src/app/Header.js [hunk 1]
```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
   )
 }
 
+function NavLinkAuth() {
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:58]: LogOutLink

File: src/app/Header.js [hunk 2]
```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
...
+      localStorage.removeItem(SIGNUP_CURRENT_STEP_KEY);
+    },
+    onError: () => {
+      console.error("Logout failed");
+    }
+  });
+
+  const theme = useTheme()
+
+  useEffect(() => {
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:73]: Same here, why should logout component **set**, i.e. perform login?

File: src/app/Header.js [hunk 3]
```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
...
+    setAuthToken(localStorage.getItem(AUTH_TOKEN_KEY));
+  }, []);
+
+  return (
+    <NextLink href="/" passHref>
+      <Button variant="outlined" color="orange" onClick={authToken ? mutation.mutate : null} sx={{
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:79]: Why do we read initial authtoken from state?
User logged in? Read auth token from localStorage, delete it.
User not logged in? Read auth token from state, do nothing.

Comment (maryna-velychko) [reply src/app/Header.js:79]: 1. it is for using useEffect for getting it (solve problem with server rendering)

File: src/app/Header.js [hunk 4]
```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
...
+        textTransform: 'none',
+        fontSize: theme.typography.htmlFontSize,
+      }}>
+      {authToken ? 'Log out': 'Sign in'}
```
[the remark this answers is not in the document]
Comment (maryna-velychko) [reply src/app/Header.js:83]: I can separate it but it is code duplication. In this case we need to check authToken and show or don't show this

File: src/app/Header.js (hunk 3 above)
Comment (maryna-velychko) [reply src/app/Header.js:79]: Do you want to show Log out botton in any case?

File: src/app/Header.js [hunk 5]
```diff
@@ -53,12 +55,44 @@ function NavLink({link}) {
   )
 }
 
+function NavLinkAuth() {
+  const [authToken, setAuthToken] = useState(null);
+  const mutation = useMutation(logout, {
+    onSuccess: () => {
+      localStorage.removeItem(AUTH_TOKEN_KEY);
+      localStorage.removeItem(USER_ID_KEY);
```
[the remark this answers is not in the document]
Comment (maryna-velychko) [reply src/app/Header.js:63]: We can delete this keys after login but as you write before, now user is not supposed to even see a logout button and we need to refresh it after logout. For example we can change logic and for login email confirmation is not need. In this case we can show users next signup step after login

File: src/app/Header.js [hunk 6]
```diff
@@ -53,12 +55,58 @@ function NavLink({link}) {
   )
 }
 
+function LogOutLink() {
+  const theme = useTheme()
+  const mutation = useMutation(logout, {
+    onSuccess: () => {
+      localStorage.removeItem(AUTH_TOKEN_KEY);
+      console.log("Logout success");
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:63]: It should go to snackbar. Seach for snackbar to see usages examples.

File: src/app/Header.js [hunk 7]
```diff
@@ -53,12 +55,58 @@ function NavLink({link}) {
...
+    </Button>
+  </NextLink>
+  );
+}
+
+function SignInLink() {
+  const theme = useTheme()
+
+  return (
+    <NextLink href="/" passHref>
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:86]: Did you test it? It does not lead to a login page judging fro href.

**Pull request merged**